internal/pkg/template/templates/environment/cf.yml

```diff
@@ -284,25 +290,52 @@ Resources:
 {{- else}}
       VpcId: !Ref VPC
 {{- end}}
-  InternalALBHTTPListener:
+  InternalHTTPListener:
     Metadata:
       'aws:copilot:description': 'An internal load balancer listener to route HTTP traffic'
     Type: AWS::ElasticLoadBalancingV2::Listener
     Condition: CreateInternalALB
     Properties:
       DefaultActions:
-        - TargetGroupArn: !Ref DefaultInternalALBHTTPTargetGroup
+        - TargetGroupArn: !Ref DefaultInternalHTTPTargetGroup
           Type: forward
       LoadBalancerArn: !Ref InternalLoadBalancer
       Port: 80
       Protocol: HTTP
+  InternalHTTPSListener:
+    Metadata:
+      'aws:copilot:description': 'An internal load balancer listener to route HTTPS traffic'
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Condition: ExportInternalHTTPSListener
+    Properties:
+      Certificates:
+{{- if .ImportCertARNs}}
+        - CertificateArn: {{index .ImportCertARNs 0}}
+{{- end}}
```

Comment:
Does this mean if there's no imported cert this one will be an empty value? I thought CFN won't like it then.

I *think* we still need the lambda for the default certificate if they don't import any certs.

Reply:
Thanks for catching this. Added a `CreateInternalHTTPSListener` parameter that is true when certs are imported and no public subnets are imported.